go/integTest/batch_test.go

```diff
@@ -184,6 +155,157 @@ func (suite *GlideTestSuite) TestWatch_and_Unwatch_cross_slot() {
+			ZRandMember(key1).
+			ZRandMemberWithCount(key1, 42).
+			ZRandMemberWithCountWithScores(key1, 42).
+			ZMScore(key1, []string{"a"}).
+			ZInter(options.KeyArray{Keys: []string{key1, key2}}).
+			ZInterWithScores(options.KeyArray{Keys: []string{key1, key2}}, *options.NewZInterOptions().SetAggregate(options.AggregateMax)).
+			ZUnion(options.KeyArray{Keys: []string{key1, key2}}).
+			ZUnionWithScores(options.KeyArray{Keys: []string{key1, key2}}, *options.NewZUnionOptionsBuilder().SetAggregate(options.AggregateMax)).
+			XAdd(key1, [][]string{{"a", "b"}}).
+			XAddWithOptions(key1, [][]string{{"a", "b"}}, *options.NewXAddOptions().SetId("0-1")).
+			XAutoClaim(key1, "g", "c", 2, "0-0").
+			XAutoClaimWithOptions(key1, "g", "c", 2, "0-0", *options.NewXAutoClaimOptions().SetCount(2)).
+			XAutoClaimJustId(key1, "g", "c", 2, "0-0").
+			XAutoClaimJustIdWithOptions(key1, "g", "c", 2, "0-0", *options.NewXAutoClaimOptions().SetCount(2)).
+			XReadGroup("g", "c", map[string]string{key1: "0-0"}).
+			XReadGroupWithOptions("g", "c", map[string]string{key1: "0-0"}, *options.NewXReadGroupOptions().SetNoAck()).
+			XRead(map[string]string{key1: "0-0"}).
+			XReadWithOptions(map[string]string{key1: "0-0"}, *options.NewXReadOptions().SetCount(2)).
+			XPending(key1, "g").
+			XPendingWithOptions(key1, "g", *options.NewXPendingOptions("0-0", "2-2", 3)).
+			XClaim(key1, "g", "c", 2, []string{"0-0"}).
+			XClaimWithOptions(key1, "g", "c", 2, []string{"0-0"}, *options.NewXClaimOptions().SetForce()).
+			XClaimJustId(key1, "g", "c", 2, []string{"0-0"}).
+			XClaimJustIdWithOptions(key1, "g", "c", 2, []string{"0-0"}, *options.NewXClaimOptions().SetForce()).
+			XInfoStream(key1).
+			XInfoStreamFullWithOptions(key1, options.NewXInfoStreamOptionsOptions().SetCount(2)).
+			XInfoConsumers(key1, "g").
+			XInfoGroups(key1).
+			XRange(key1, options.NewStreamBoundary("0-0", true), options.NewStreamBoundary("2-0", true)).
+			XRangeWithOptions(key1, options.NewStreamBoundary("0-0", true), options.NewStreamBoundary("2-0", true), *options.NewXRangeOptions().SetCount(2)).
+			XRevRange(key1, options.NewStreamBoundary("0-0", true), options.NewStreamBoundary("2-0", true)).
+			XRevRangeWithOptions(key1, options.NewStreamBoundary("0-0", true), options.NewStreamBoundary("2-0", true), *options.NewXRangeOptions().SetCount(2))
+
+		if suite.serverVersion >= "7.0.0" {
+			transaction.
+				ZMPop([]string{key1}, constants.MAX).
+				ZMPopWithOptions([]string{key1}, constants.MAX, *options.NewZMPopOptions().SetCount(2)).
+				LMPop([]string{key1}, constants.Left).
+				LMPopCount([]string{key1}, constants.Left, 42)
+		}
+		if suite.serverVersion >= "7.2.0" {
+			transaction.
+				ZRankWithScore(key1, "d").
+				ZRevRankWithScore(key1, "d")
+		}
+
+		res, err := runBatchOnClient(client, transaction, false, nil)
+		suite.NoError(err)
+		for i, resp := range res {
+			suite.Equal("WRONGTYPE: Operation against a key holding the wrong kind of value", resp.(error).Error(), i)
```

Comment:
This will panic if `resp` cannot be converted to an error.  In Go tests, any panic will stop the tests an exit out.